File: java/data/Player.java
package data;

import data.Pieces;

public class Player {
	private static final int howManyPieces=12;

	private int piecesNumbers;
	private int blockedPiecesNumbers;
	private int[][] getMove=new int[2][2];
	private Pieces[] piecesPosition;

	public int getHowManyPieces(){
		return howManyPieces;
	}

	public int getPiecesNumbers() {
		return piecesNumbers;
	}

	public int getBlockedPiecesNumbers() {
		return blockedPiecesNumbers;
	}

	public Pieces getPieces(int number){
		return piecesPosition[number];
	}

	public int getGetMove(int number1, int number2) {
		return getMove[number1][number2];
	}

	public void setGetMove(int[][] getMove) {
		this.getMove = getMove;
	}

	public Player(){
		int i;
		this.piecesPosition=new Pieces[howManyPieces+1];
		for(i=0; i<=howManyPieces; i++){
			this.piecesPosition[i]=new Pieces();
		}
		//Pawn which number 0 don't exist;
		this.piecesPosition[0].setPosition(-1);
		this.piecesPosition[0].setCanMove(false);
		this.piecesPosition[0].setOnBeating(false);
	}

	public void countBlockedAndAllPieces(){
		int i;
		Pieces piece=null;

		this.blockedPiecesNumbers=0;
		this.piecesNumbers=0;
		for(i=1; i<=howManyPieces; i++){
			piece=getPieces(i);
			if(piece.getPosition()!=-1){
				this.piecesNumbers++;
				if(!piece.isCanMove()){
					this.blockedPiecesNumbers++;
				}
			}
		}
	}

	public void testPlayer(){
		int i;
		for(i=0; i<(howManyPieces+1); i++){
			System.out.println(i+">Position:"+this.piecesPosition[i].getPosition()
					+" >canMove:"+this.piecesPosition[i].isCanMove()
					+" >onBeating:"+this.piecesPosition[i].isOnBeating());
		}
	}

}

File: C/player.cpp
#include "player.h"

void GetHumanMove(int GetMove[2][2], int pos_x, int pos_y, int x)
{
	if((pos_x > 0) && (pos_x < 75)){
		GetMove[x][1] = 1;
	}
	if((pos_x > 75) && (pos_x < 150)){
		GetMove[x][1] = 2;
	}
	if((pos_x > 150) && (pos_x < 225)){
		GetMove[x][1] = 3;
	}
	if((pos_x > 225) && (pos_x < 300)){
		GetMove[x][1] = 4;
	}
	if((pos_x > 300) && (pos_x < 375)){
		GetMove[x][1] = 5;
	}
	if((pos_x > 375) && (pos_x < 450)){
		GetMove[x][1] = 6;
	}
	if((pos_x > 450) && (pos_x < 525)){
		GetMove[x][1] = 7;
	}
	if((pos_x > 525) && (pos_x < 600)){
		GetMove[x][1] = 8;
	}
	
	if((pos_y > 0) && (pos_y < 75)){
		GetMove[x][0] = 1;
	}
	if((pos_y > 75) && (pos_y < 150)){
		GetMove[x][0] = 2;
	}
	if((pos_y > 150) && (pos_y < 225)){
		GetMove[x][0] = 3;
	}
	if((pos_y > 225) && (pos_y < 300)){
		GetMove[x][0] = 4;
	}
	if((pos_y > 300) && (pos_y < 375)){
		GetMove[x][0] = 5;
	}
	if((pos_y > 375) && (pos_y < 450)){
		GetMove[x][0] = 6;
	}
	if((pos_y > 450) && (pos_y < 525)){
		GetMove[x][0] = 7;
	}
	if((pos_y > 525) && (pos_y < 600)){
		GetMove[x][0] = 8;
	}
}
